[SRVKS-1115] Performance and scalability of OpenShift Serverless Serving (#80197)
openshift/openshift-docs: modules/serverless-minimal-requirements-serving.adoc

Reviewer: not highlighting each one from here on, but replace e.g "OpenShift Serverless" with {ServerlessProductName} in here and throughout the files in this PR. Same with "OpenShift"/"OpenShift Container Platform" and {product-title}.

// Module included in the following assemblies:
//
// * /knative-serving/scalability-and-performance-serving.adoc

:_mod-docs-content-type: CONCEPT
[id="serverless-minimal-requirements-serving_{context}"]
= Minimal requirements of OpenShift Serverless Serving

> While the default set-up is suitable for medium-sized workloads, it might be over-sized for smaller set-ups or under-sized for high-workload scenarios.

Reviewer 2: ```suggestion
While the default setup is suitable for medium-sized workloads, it might be over-sized for smaller setups or under-sized for high-workload scenarios.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#s

To configure OpenShift Serverless Serving for a minimal workload scenario, you need to know the idle consumption of the system components.

[id="serverless-minimal-requirements-serving-idle-consumption_{context}"]
== Idle consumption

> The idle consumption is dependent on the number of Knative Services. The following memory utilization has been measured for the components in the `knative-serving` and `knative-serving-ingress` OpenShift projects:

Reviewer 2: ```suggestion
The idle consumption is dependent on the number of Knative Services. The following memory usage has been measured for the components in the `knative-serving` and `knative-serving-ingress` OpenShift projects:
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#u

[cols=5*,options="header"]
|===

> |Component

Reviewer: The component names are in backticks usually, probably should be in this first column as well?

|0 Services
|100 Services
|500 Services
|1000 Services

|activator
|55Mi
|86Mi
|300Mi
|450Mi

|autoscaler
|52Mi
|102Mi
|225Mi
|350Mi

|controller
|100Mi
|135Mi
|310Mi
|500Mi

|webhook
|60Mi
|60Mi
|60Mi
|60Mi

|3scale-kourier-gateway
|20Mi
|60Mi
|190Mi
|330Mi

|net-kourier-controller
|90Mi
|170Mi
|340Mi
|430Mi

|===

[NOTE]
====
Either `3scale-kourier-gateway` and `net-kourier-controller` components or `istio-ingressgateway` and `net-istio-controller` components are installed. 

> The memory consumption of `net-istio` will be based on the total number of pods within the mesh.

Reviewer: ```suggestion
The memory consumption of `net-istio` is based on the total number of pods within the mesh.
```

// Module included in the following assemblies:
//
// * /knative-serving/scalability-and-performance-serving.adoc

:_content-type: CONCEPT
[id="serverless-minimal-requirements-serving_{context}"]
= Minimal requirements of OpenShift Serverless Serving

While the default set-up is suitable for medium-sized workloads, this might be over-sized for smaller set-ups or under-sized for high-workload scenarios.
To configure OpenShift Serverless Serving for a minimal workload scenario, you need to know the idle consumption of the system components.

[id="serverless-minimal-requirements-serving-idle-consumption_{context}"]
== Idle consumption

The idle consumption is dependent on the number of Knative Services. The following memory utilization has been measured for the components in the `knative-serving` and `knative-serving-ingress` OpenShift projects:

[cols=5*,options="header"]
|===
|Component
|0 Services

> |100 services

Reviewer 3: ```suggestion
|100 Services
```

// Module included in the following assemblies:
//
// * /knative-serving/scalability-and-performance-serving.adoc

:_content-type: CONCEPT
[id="serverless-minimal-requirements-serving_{context}"]
= Minimal requirements of OpenShift Serverless Serving

While the default set-up is suitable for medium-sized workloads, this might be over-sized for smaller set-ups or under-sized for high-workload scenarios.
To configure OpenShift Serverless Serving for a minimal workload scenario, you need to know the idle consumption of the system components.

[id="serverless-minimal-requirements-serving-idle-consumption_{context}"]
== Idle consumption

The idle consumption is dependent on the number of Knative Services. The following memory utilization has been measured for the components in the `knative-serving` and `knative-serving-ingress` OpenShift projects:

> * 170 Knative Services

Reviewer 3: This looks like it mixes two separate measurements.  If this is about *idle* behavior with existence of 0/100/500/1000 idle ksvcs, than this list doesn't make sense and should be removed.
